src/odemis/driver/pwrmccdaq.py

```diff
@@ -427,16 +429,13 @@ def DOut(self, port_number, value):
         if value < 0 or value > 255:
             raise ValueError("Value to set is not between 0 and 255")
 
-        DIO_port = self._return_port_status(port_number, read=False)
-
-        if port_number == usb_1208LS.DIO_PORTA:
-            bit_config = self.port_a_bit_config
-        elif port_number == usb_1208LS.DIO_PORTB:
-            bit_config = self.port_b_bit_config
-        else:
-            raise ValueError()
+        DIO_port = self.port_status[port_number]
+        bit_config = self.port_config[port_number]
+        DIO_port[:] = [bool(value & 1 << i) if not bit_config[i] else DIO_port[i] for i in range(8)]
 
-        # when port A is requested, writing to port B is simulated and the other way around
+        # Simulate connection to the other port
+        other_port = usb_1208LS.DIO_PORTA if port_number == usb_1208LS.DIO_PORTB else usb_1208LS.DIO_PORTB
+        DIO_port = self.port_status[other_port]
```

Comment:
Line 438 is overriding the value of line 434. I am not sure if that is intentional.

Reply:
Yes, it's the same code, to show we do the same thing (for simulation).